hack/observability/grafana/dashboards/cluster-api-performance.json

```diff
@@ -0,0 +1,5078 @@
+{
```

Comment:
New dashboard to visualize the user-facing performance

![pr-perf](https://github.com/kubernetes-sigs/cluster-api/assets/4662360/31ca4caa-a802-4a6f-82a7-f7edebcc9e73)